pdns/lua-record.cc

```diff
@@ -607,18 +607,34 @@ static void setupLuaRecords()
     });
   lua.writeFunction("createForward", []() {
       DNSName rel=s_lua_record_ctx->qname.makeRelative(s_lua_record_ctx->zone);
+      // parts is something like ["1", "2", "3", "4", "static"] or
+      // ["1", "2", "3", "4"] or ["ip40414243", "ip-addresses", ...]
       auto parts = rel.getRawLabels();
-      if(parts.size()==4)
+      // Yes, this still breaks if an 1-2-3-4.XXXX is nested too deeply...
+      if(parts.size()>=4) {
         return parts[0]+"."+parts[1]+"."+parts[2]+"."+parts[3];
-      if(parts.size()==1) {
+      } else if (parts.size() >= 1) {
         // either hex string, or 12-13-14-15
-        //        cout<<parts[0]<<endl;
+        vector<string> ip_parts;
+        stringtok(ip_parts, parts[0], "-");
         unsigned int x1, x2, x3, x4;
-        if(sscanf(parts[0].c_str()+2, "%02x%02x%02x%02x", &x1, &x2, &x3, &x4)==4) {
+        if (ip_parts.size() >= 4) {
+          // 1-2-3-4 with any prefix (e.g. ip-foo-bar-1-2-3-4)
+          string ret;
+          for (size_t n=4; n > 0; n--) {
+            auto octet = std::stoll(ip_parts[ip_parts.size() - n]);
+            if (octet > 0 && octet <= 255) {
+              ret += ip_parts[ip_parts.size() - n] + ".";
```

Comment:
I wonder if it would be cleaner/easier to read if you reused `octet` here (which would mean converting it back to a string).

Reply:
I'm fine either way, but not doing _another_ conversion would be nice :)

Reply:
```
auto octet = ip_parts[ip_parts.size() - n];
auto octetVal = std::stol(octet);
...
```
Also `std::stol` can throw `std::invalid_argument` or std::out_of_raneg`